plugin/mouse/mouse.py

```diff
@@ -247,24 +249,29 @@ def show_cursor_helper(show):
         ctrl.cursor_visible(show)
 
 
-def on_pop(_active):
-    # Only want the pop noise to click when we're using an eye tracker
-    is_using_eye_tracker = (
-        actions.tracking.control_zoom_enabled()
-        or actions.tracking.control_enabled()
-        or actions.tracking.control1_enabled()
-    )
-
+@ctx.action("user.noise_trigger_pop")
+def on_pop():
     if setting_mouse_enable_pop_stops_scroll.get() >= 1 and (gaze_job or scroll_job):
+        # Allow pop to stop scroll
         stop_scroll()
-    elif is_using_eye_tracker and not actions.tracking.control_zoom_enabled():
-        if setting_mouse_enable_pop_click.get() >= 1:
+    else:
+        # Otherwise respect the mouse_enable_pop_click setting
+        setting_val = setting_mouse_enable_pop_click.get()
+
+        is_using_eye_tracker = (
+            actions.tracking.control_zoom_enabled()
+            or actions.tracking.control_enabled()
+            or actions.tracking.control1_enabled()
+        )
+        should_click = setting_val == 2 or (
```

Comment:
I think mode two (2) should still exclude the zoom mouse, as it will cause nonobvious problems like double clicking or dismissing the zoom mouse. maybe add a warning?

otherwise, looks good to me

Reply:
I've not used an eye tracker. Are they something like this?

* Control mouse: just move cursor with eyes (and head).
* Zoom mouse: move cursor with eyes (and head). Pop to zoom in, then pop to click while zoomed in. After clicking return to unzoomed state (like https://www.youtube.com/watch?v=VMNsU7rrjRI)

For the zoom mouse case, shouldn't people be using this setting with value = 1 (or indeed 0)? I'm thinking option 2 is mostly for people who want to position the cursor with the mouse or something like that, but don't want to click with their finger.

Reply:
Yeah, those descriptions are accurate. 

The issue is zoom mouse registers its own pop handler that the user doesn't currently control.

As such, I would personally vote to err on the side of making it hard to misconfigure.

Reply:
Fair enough, I'll add an exclusion for zoom mouse to option 2.